### PR TITLE
Disable actionlint hook by default when running pre-commit locally

### PR DESCRIPTION
**Pull request by AlexWaygood**
## Summary

We added actionlint as a pre-commit hook in https://github.com/astral-sh/ruff/pull/15021. Unfortunately, it's by far the slowest hook in our pre-commit config, so this now makes running `uvx pre-commit run -a` locally quite slow.

This PR tweaks our pre-commit config so that it will now be disabled by default when running pre-commit locally but will be enabled when pre-commit is run in CI. This is achieved by reworking the hook entry so that it is only run when `--hook-stage=manual` is specified on the command line.

## Test Plan

The hook now runs instantaneously when `uvx pre-commit run -a` is invoked for me locally (because it is skipped), but still picks up errors in GitHub Actions workflows when `--hook-stage=manual` is specified. I also checked the exact command we'll be running in `.github/workflows/ci.yaml`, and actionlint correctly flags issues locally when that command is run.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -585,7 +585,7 @@ jobs:
         run: |
           echo '```console' > "$GITHUB_STEP_SUMMARY"
           # Enable color output for pre-commit and remove it for the summary
-          SKIP=cargo-fmt,clippy,dev-generate-all pre-commit run --all-files --show-diff-on-failure --color=always | \
+          SKIP=cargo-fmt,clippy,dev-generate-all pre-commit run --all-files --show-diff-on-failure --color=always --hook-stage=manual | \
```
Comment (MichaReiser) [inline .github/workflows/ci.yaml:588]: Nit: It wasn't very clear to me what *manual* means. What would you think of renaming it to `CI`?

**Review by MichaReiser — APPROVED** [review]
Thank you!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (AlexWaygood) [reply .github/workflows/ci.yaml:588]: Unfortunately this is out of my control 😆 it has to be one of git's recognised hook stages or `manual`: https://pre-commit.com/#confining-hooks-to-run-at-certain-stages

Comment (AlexWaygood) [reply .github/workflows/ci.yaml:588]: I'll add a comment to this workflow file, though

**Pull request merged**